### ISSUE TITLE
Enable dynamic pruning immediately

### ISSUE THREAD

**Issue by jpountz** [Tony-X/search-benchmark-game#41]
This benchmark uses `IndexSearcher#search(Query, int)` which will only start to dynamically prune after collecting 1,000 hits. This can be fixed by creating the collector manually.

```
TopScoreDocsCollector collector = TopScoreDocCollector.create(10, 10); // second argument is now 10 instead of 1,000
searcher.search(query, collector);
```

**Comment by mikemccand** [comment]
Oh good catch @jpountz!  Is 10 what Tantivy effectively does?

**Comment by jpountz** [comment]
I don't know what Tantivy does, I was just assuming given the name of the task that computing a partial hit count was out of scope. Maybe Tantivy does some counting too.

**Comment by Tony-X** [comment]
@jpountz thanks for taking a look :) 

Your observation is correct - I believe that code live is for the TOP_10 task (skipping allowed) where your proposal is more precise. Although currently, the benchmark runs only `TOP_COUNT` and `COUNT` commands just to remove one variable out of the table (skipping with BMW). 

Nevertheless, we should fix this. Feel free to raise a PR!

**Comment by fulmicoton** [comment]
Nice catch! What is the rationale for pruning only after 1000 hits by default?

> Is 10 what Tantivy effectively does?

Tantivy prunes as soon as possible for collectors where it makes sense.

**Comment by jpountz** [comment]
The rationale is to help users implement old-school pagination where you have a link for each page of results up to some reasonable number like 10 or 20 pages at most. To do that you need to know whether there is a n-th page or not, in order to know whether to add a link in the UI. The value of 1,000 was chosen because the overhead should be quite small, and it should be enough most of the time.

That said I don't know how commonly this feature is actually used. It could be that we are adding some overhead to all top-k searches and nobody takes advantage of this partial hit count information.

**Comment by slow-J** [comment]
Thanks Adrien for the catch!
I'll patch this since this code will now be live with the changes for enabling TOP_100 in https://github.com/Tony-X/search-benchmark-game/issues/2.
